### ISSUE TITLE
False positive: NullAway fails to infer non-nullability from `@Contract("null -> false")` variable assignment in complex boolean expression

### ISSUE THREAD

**Issue by akbarkanso** [uber/NullAway#1443]
I am encountering a false positive where NullAway reports a `passing @Nullable parameter where @NonNull is required` error, despite the variable being guarded by a utility method annotated with `@Contract("null -> false")`.

The issue seems to arise when the result of the contract method is assigned to a boolean variable which is then used in an `if` condition, rather than calling the method directly inside the `if` statement.

### **Minimal Reproducible Example**

**1. The Utility Class (with Contract)**

```java
import org.jetbrains.annotations.Contract;
import javax.annotation.Nullable;

public class ServiceUtils {
    // Mimics a library method checking for text presence
    @Contract("null -> false")
    public static boolean isValidString(@Nullable String value) {
        return value != null && !value.isEmpty();
    }
}

```

**2. The Class triggering the error**

```java
import javax.annotation.Nullable;
import javax.annotation.CheckForNull;
import java.util.Objects;

public class ItemProcessor {

    @Nullable
    private String metadata;

    @CheckForNull
    public String getMetadata() {
        return metadata;
    }

    public void process() {
        // 1. Get nullable value
        var currentMetadata = getMetadata();

        // 2. Boolean logic using the Contract method
        // implied: if hasSpecificMetadata is true, currentMetadata cannot be null
        var hasSpecificMetadata = ServiceUtils.isValidString(currentMetadata) 
                                  && !"EXCLUDE".equals(currentMetadata);

        if (hasSpecificMetadata) {
            // ERROR HERE: NullAway considers currentMetadata to be @Nullable here
            consumeNonNull(currentMetadata); 
        }
    }

    public void consumeNonNull(@NonNull String input) {
        // do something
    }
}

```

### **Expected Behavior**

NullAway should recognize that because `ServiceUtils.isValidString(currentMetadata)` is required to be `true` for execution to enter the `if` block, and that method has the contract `null -> false`, `currentMetadata` is guaranteed to be non-null inside the block.

### **Actual Behavior**

NullAway fails the build with:

```text
error: [NullAway] passing @Nullable parameter 'currentMetadata' where @NonNull is required
consumeNonNull(currentMetadata);

```

### **Workaround**

Explicitly unwrapping the variable satisfies NullAway, confirming that the tool understands the type, but misses the flow analysis implication:

```java
// This works
consumeNonNull(Objects.requireNonNull(currentMetadata));

```

### **Environment**

* **NullAway Version:** 0.12.9
* **Java Version:** 21
* **Build System:** Gradle
